### PR TITLE
(MODULES-2637) Checks java actually installed

### PR DESCRIPTION
**Pull request by petems**
The other method for running `which java` doesn't work on OSX, as java is installed as an empty shim when not installed in the base OSX image:

```
vagrant:~ vagrant$ java
No Java runtime present, requesting install.
vagrant:~ vagrant$ /usr/libexec/java_home --failfast 
Unable to find any JVMs matching version "(null)".
vagrant:~ vagrant$ 
```

We can use the `/usr/libexec/java_home --failfast` command to check if java is actually present first

### CODE REVIEW

**Comment by tphoney** [comment]
@petems thanks for this, but you are getting unit test failures
Failure/Error: Facter.value(:java_version).should be_nil
     Mocha::ExpectationError:
       unexpected invocation: Facter::Util::Resolution.exec('java -Xmx8m -version 2>&1')

File: spec/unit/facter/java_version_spec.rb [hunk 1]
```diff
@@ -23,6 +23,22 @@
           Facter.value(:java_version).should == "1.7.0_71"
         end
       end
+      context 'on Darwin' do
+        before do
+          Facter.fact(:operatingsystem).stubs(:value).returns("Darwin")
+        end
+        let(:facts) { {:operatingsystem => 'Darwin'} }
```
Comment (DavidS) [inline spec/unit/facter/java_version_spec.rb]: L30 here and L27-L29 above should do the same, you can drop the `before` block.

**Comment by DavidS** [comment]
Even running the test on its own leads to this failure. facter seems to fall back to the unconfined fact if your version is returning `nil`; adding `else 'none'` to the Darwin's fact's `unless` makes the test fail at the expected `nil != "none".

**Comment by DavidS** [comment]
@petems I updated your commit (https://github.com/puppetlabs/puppetlabs-java/compare/master...DavidS:MODULES-2637-java_version_on_OSX) with something that now passes the unit tests, please have a look at that

**Comment by petems** [comment]
Merged in https://github.com/puppetlabs/puppetlabs-java/pull/142
